plugin/buycourses/src/function.php

```diff
@@ -230,54 +222,40 @@
 }
 
 if ($_REQUEST['tab'] == 'courses_filter') {
-    $course = Database::escape_string($_REQUEST['course']);
-    $priceMin = Database::escape_string($_REQUEST['pricemin']);
-    $priceMax = Database::escape_string($_REQUEST['pricemax']);
+    $course = isset($_REQUEST['name']) ? Database::escape_string($_REQUEST['name']) : '';
+    $priceMin = isset($_REQUEST['pricemin']) ? floatval($_REQUEST['pricemin']) : 0;
+    $priceMax = isset($_REQUEST['pricemax']) ? floatval($_REQUEST['pricemax']) : 0;
     /**
      * Deprecated since 2014-10-14
      */
     //$show = Database::escape_string($_REQUEST['show']);
-    $category = Database::escape_string($_REQUEST['category']);
+    //$category = Database::escape_string($_REQUEST['category']);
     $server = api_get_path(WEB_PATH);
 
-    $filter = '';
-    if ($course != '') {
-        $filter .= "b.title LIKE '%" . $course . "%'";
-    }
-    if ($priceMin != '') {
-        if ($filter == '') {
-            $filter .= "a.price >= '" . $priceMin . "'";
-        } else {
-            $filter .= " AND a.price >= '" . $priceMin . "'";
-        }
+    $sql = "SELECT a.course_id, a.visible, a.price, b.*
+            FROM $tableBuyCourse a, $tableCourse b
+            WHERE a.course_id = b.id AND a.session_id = 0
+            AND a.visible = 1;";

```

Comment:
No need to include ";" inside the query string. Just saying. No need to fix, but in general, it is unnecessary.